app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,103 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.where(complete: !true)
+    @completed_tasks = Task.where(complete: true)
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    @task = Task.new
+    unless params["task"]
+      render :new, status: :bad_request
+      return
+    end
+
+    @task.name = params[:task][:name]
+    @task.description = params[:task][:description]
+    if params[:task][:complete_by]
+      @task.complete_by = Date.parse(params[:task][:complete_by])
```

Comment:
Just a note, if the user submits an empty string for complete_by, this will raise an error.  It would be good to rescue this.